pageserver/src/tenant/mgr.rs

```diff
@@ -1271,10 +1269,13 @@ impl TenantManager {
 
 #[derive(Debug, thiserror::Error)]
 pub(crate) enum GetTenantError {
+    /// NotFound is a TenantId rather than TenantShardId, because this error type is used from
+    /// getters that use a TenantId and a ShardSelector, not just getters that target a specific shard.
     #[error("Tenant {0} not found")]
     NotFound(TenantId),
```

Comment:
Would be nice to include the ShardSelector in the error and error message

Reply:
It's a bit awkward: there are some cases where we return NotFound because a selector didn't match, and some where we return NotFound because a shard was present but not active, etc.  If it was a drop-in change I'd do it inline, but I think it would need a broader pass over the error handling in these paths

Reply:
> but I think it would need a broader pass over the error handling in these paths

+1 on that